SMSegmentViewController/SMSegmentView/SMSegmentView.swift

```diff
@@ -124,7 +124,6 @@ open class SMSegmentView: UIControl {
         self.segments.insert(segment, at: index)
 
         self.addSubview(segment)
```

Comment:
I believe `setNeedsLayout()` is needed here as well, in order for the segments to resize accordingly. Same applies to the following function to remove a segment.

Reply:
Nope! `-addSubview:` invalidates the layout already, there's no need to manually call `-setNeedsLayout`

Reply:
Ah, yes, that's true. Thanks!